### ISSUE TITLE
Consider adding Stencil to the benchmark

### ISSUE THREAD

**Issue by LifeIsStrange** [krausest/js-framework-benchmark#860]
I'm aware that it was mentioned in that closed issue:
https://github.com/krausest/js-framework-benchmark/issues/441
And I understand that you expect external contributors for adding frameworks as there are so many!

However, while (as you know) there is a HUGE number of js "frameworks" out there, I have taken a look at all the missing frameworks referenced in that issue and Stencil *really* stands out:
Here's my analysis of why:
From that list there are three frameworks that have a decent number of commits (and therefore expected maturity):
Omi:
https://github.com/Tencent/omi
But it has a very low number of users ~100
Skatejs:
https://github.com/skatejs/skatejs
But it's development has stopped in 2019
And
Stencil:
https://stenciljs.com
Which has ~10000 users on github which for reference is 3 times as much as infernojs.
I would say that it is the 5th most popular vdom framework.
It also is used by Apple and Microsoft!
And it has a huge downstream user: the Ionic framework (which is the Electron of smartphones). Stencil has a similar value proposition than Svelte: it compiles to web-components and therefore allow Ionic to implement once a widget and make it usable in any framework. In addition to the interoperability, this compilation ability usually allow next gen performance (Svelte is the fastest of the 4th popular Vdoms for this reason) and benchmarking Stencil would be very interesting as it could be even faster than svelte and therefore push the boundaries of what a production ready js framework can do! 
I also think that it would increase its popularity because having the performance of a framework as an unknown is a major risk.

Hence why I hope that you agree that Stencil support should be prioritarized and that letting this issue open would help give it visibility.

**Comment by ryansolid** [comment]
Stencil does really standout as being absent. The thing I've found though is it takes a certain expertise in the framework to be able to write these benchmarks in a way that is representative. Everyone that I've added I've pretty much needed to talk to the author, core maintainer about details to make sure I was doing the right thing.

People write these issues, but very rarely does it result in new submissions. I've had the best luck going to the source repo and asking there. This is where I'd be putting my focus: https://github.com/ionic-team/stencil/issues/2765

I did see this: https://github.com/ionic-team/stencil/issues/2716. But I disagree strongly with the response. The problem is while I don't want to put this on the maintainers, it's the like RealWorld demo or TodoMVC except more stringent as we are showcasing performance.

I feel this benchmark only works because Evan worked on the Vue implementation, Rich worked on the Svelte one, Dan weighed in on the React Hooks one. You see the same for the Ember community, Inferno, pretty much every library here has had the input of core contributors or maintainers. That's the only way we get good results. Benchmarks are hard. But on the flip side, it's the reason (besides @krausest tireless work) that I have so much confidence in the results.

Now I'm sure if someone started it for them and asked for input that would probably expedite this.  But I think the best people to do that are probably in the Stencil community.

I did look briefly at making a Stencil implementation a year back but I didn't feel comfortable (or ready to make the time commitment).  Stencil is a difficult one for me because of its WebComponent nature. I think we can probably make the whole benchmark a single WebComponent like the LitElement one, but it would take more work to test the different breaks up to decide what is best. With VDOM libraries often the component boundaries have performance implications on re-renders so there are tradeoffs on how you break things up. Secondly I'd worry about doing things the idiomatic way versus the most cutthroat. Even something I'm pretty familiar with like updating from Svelte 2 to Svelte 3 I had some reservations on the most representative approach, and without buy-in I didn't bother putting in all the optimizations I could (this is before we started marking implementations for those sort of things). But it still stands. And this was just updating something that Rich had worked on previously.

Mind you if anyone wants tips on how to get good performance in this benchmark feel free to tag me in your specific repo issues. That is an area I feel I could be of great help.
